### PR TITLE
Added hsrp event history parsers (from PR #555)

### PR DESCRIPTION
**Pull request by divya883**
## Description
Added parsers for the commands 'show hsrp internal event-history errors', 'show hsrp internal event-history msgs' and 'show hsrp internal event-history debugs' for nxos.

## Motivation and Context
I was working with hsrp as part of a project and wanted to retrieve related info and found that there were no parsers for the above commands.

## Impact (If any)
No negative impact expected.

## Screenshots:
show hsrp internal event-history errors

2021-12-14T19:09:08: %AETEST-INFO: +----------------------------------------------------------+
2021-12-14T19:09:08: %AETEST-INFO: |                       STEPS Report                       |
2021-12-14T19:09:08: %AETEST-INFO: +----------------------------------------------------------+
2021-12-14T19:09:08: %AETEST-INFO: STEP 1 - nxos -> ShowHsrpEventHistoryErrors           Passed
2021-12-14T19:09:08: %AETEST-INFO: STEP 1.1 - Test Golden -> nxos -> ShowHsrpEventHistoryErrors    Passed
2021-12-14T19:09:08: %AETEST-INFO: STEP 1.1.1 - Gold -> nxos -> ShowHsrpEventHistoryErrors -> golden_output1    Passed
2021-12-14T19:09:08: %AETEST-INFO: STEP 1.2 - Test Empty -> nxos -> ShowHsrpEventHistoryErrors    Passed
2021-12-14T19:09:08: %AETEST-INFO: STEP 1.2.1 - Empty -> nxos -> ShowHsrpEventHistoryErrors -> empty_output    Passed
2021-12-14T19:09:08: %AETEST-INFO: ------------------------------------------------------------

show hsrp internal event history msgs

2021-12-14T19:08:56: %AETEST-INFO: +----------------------------------------------------------+
2021-12-14T19:08:56: %AETEST-INFO: |                       STEPS Report                       |
2021-12-14T19:08:56: %AETEST-INFO: +----------------------------------------------------------+
2021-12-14T19:08:56: %AETEST-INFO: STEP 1 - nxos -> ShowHsrpEventHistoryMsgs             Passed
2021-12-14T19:08:56: %AETEST-INFO: STEP 1.1 - Test Golden -> nxos -> ShowHsrpEventHistoryMsgs    Passed
2021-12-14T19:08:56: %AETEST-INFO: STEP 1.1.1 - Gold -> nxos -> ShowHsrpEventHistoryMsgs -> golden_output1    Passed
2021-12-14T19:08:56: %AETEST-INFO: STEP 1.2 - Test Empty -> nxos -> ShowHsrpEventHistoryMsgs    Passed
2021-12-14T19:08:56: %AETEST-INFO: STEP 1.2.1 - Empty -> nxos -> ShowHsrpEventHistoryMsgs -> empty_output    Passed

show hsrp event history debugs

2021-12-14T19:09:20: %AETEST-INFO: +----------------------------------------------------------+
2021-12-14T19:09:20: %AETEST-INFO: |                       STEPS Report                       |
2021-12-14T19:09:20: %AETEST-INFO: +----------------------------------------------------------+
2021-12-14T19:09:20: %AETEST-INFO: STEP 1 - nxos -> ShowHsrpEventHistoryDebugs           Passed
2021-12-14T19:09:20: %AETEST-INFO: STEP 1.1 - Test Golden -> nxos -> ShowHsrpEventHistoryDebugs    Passed
2021-12-14T19:09:20: %AETEST-INFO: STEP 1.1.1 - Gold -> nxos -> ShowHsrpEventHistoryDebugs -> golden_output1    Passed
2021-12-14T19:09:20: %AETEST-INFO: STEP 1.2 - Test Empty -> nxos -> ShowHsrpEventHistoryDebugs    Passed
2021-12-14T19:09:20: %AETEST-INFO: STEP 1.2.1 - Empty -> nxos -> ShowHsrpEventHistoryDebugs -> empty_output    Passed
2021-12-14T19:09:20: %AETEST-INFO: ------------------------------------------------------------

## Checklist:
- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

File: src/genie/libs/parser/nxos/show_hsrp_event_history.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+# ====================================================
+class ShowHsrpEventHistoryErrors(ShowHsrpEventHistoryErrorsSchema):
+
+    ''' Parser for "show hsrp internal event-history errors"'''
+
+    cli_command = 'show hsrp internal event-history errors'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
```
Comment (LukasMcClelland) [inline src/genie/libs/parser/nxos/show_hsrp_event_history.py:46]:
```suggestion
            output = self.device.execute(self.cli_command)
```

File: src/genie/libs/parser/nxos/show_hsrp_event_history.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+        event_history_errors_dict={}
+        #[2] 2021 Apr 21 12:28:21.913775 [hsrp_engine] E_DEBUG    [24399]:(Acast) Get Bundle For IOD Failed: Reason='No bundle found for VLAN', AddrType=IPv4, VLAN=1001
+        p1 = re.compile(r'^\[(?P<log_num>\d+)\]\s+(?P<date>\d{4} [a-zA-Z]+ [0-9]{1,2}) (?P<time>[0-9]{1,2}:[0-9]{1,2}:[0-9]{1,2}\.[0-9]{6}) \[(?P<proc_name>.+)\] E_DEBUG\s+\[(?P<pid>\d+)\]:(?P<msg>.+)$')
+
+        for line in out.splitlines():
```
Comment (LukasMcClelland) [inline src/genie/libs/parser/nxos/show_hsrp_event_history.py:52]:
```suggestion
        for line in output.splitlines():
```

File: src/genie/libs/parser/nxos/show_hsrp_event_history.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+# ====================================================
+class ShowHsrpEventHistoryDebugs(ShowHsrpEventHistoryDebugsSchema):
+
+    ''' Parser for "show hsrp internal event-history debugs"'''
+
+    cli_command = 'show hsrp internal event-history debugs'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
```
Comment (LukasMcClelland) [inline src/genie/libs/parser/nxos/show_hsrp_event_history.py:105]:
```suggestion
            output = self.device.execute(self.cli_command)
```

File: src/genie/libs/parser/nxos/show_hsrp_event_history.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+        event_history_debugs_dict={}
+
+        #[1] 2021 Apr 30 05:43:16.822827 [hsrp_engine] E_DEBUG    [24399]:[0]:  Time Taken For Show Run: Time=0.015553
+        p1 = re.compile(r'^\[(?P<log_num>\d+)\]\s+(?P<date>\d{4} [a-zA-Z]+ [0-9]{1,2}) (?P<time>[0-9]{1,2}:[0-9]{1,2}:[0-9]{1,2}\.[0-9]{6}) \[(?P<proc_name>.+)\] E_DEBUG\s+\[(?P<pid>\d+)\]:\[\d+\]:\s*(?P<msg>.+)$')
+
+        for line in out.splitlines():
```
Comment (LukasMcClelland) [inline src/genie/libs/parser/nxos/show_hsrp_event_history.py:112]:
```suggestion
        for line in output.splitlines():
```

File: src/genie/libs/parser/nxos/show_hsrp_event_history.py [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+# ====================================================
+class ShowHsrpEventHistoryMsgs(ShowHsrpEventHistoryMsgsSchema):
+
+    ''' Parser for "show hsrp internal event-history msgs"'''
+
+    cli_command = 'show hsrp internal event-history msgs'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
```
Comment (LukasMcClelland) [inline src/genie/libs/parser/nxos/show_hsrp_event_history.py:173]:
```suggestion
            output = self.device.execute(self.cli_command)
```

File: src/genie/libs/parser/nxos/show_hsrp_event_history.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+        #[1] 2021 May 10 12:08:54.364555 [hsrp_engine] E_MTS_RX    [REQ] Opc:MTS_OPC_SDWRAP_DEBUG_DUMP(1530), Id:0X0A8359C5, Ret:SUCCESS
+        p1_2 = re.compile(r'^\[(?P<log_num>\d+)\]\s+(?P<date>\d{4} [a-zA-Z]+ [0-9]{1,2}) (?P<time>[0-9]{1,2}:[0-9]{1,2}:[0-9]{1,2}\.[0-9]{6}) \[(?P<proc_name>[^\]]+)\] (?P<msg_type>E_MTS_RX|E_MTS_TX)\s+\[(REQ|NOT)\] Opc:(?P<opcode>[^,]+), Id:(?P<opcode_id>[^,]+), Ret:(?P<ret_val>.+)$')
+        
+        #Src:0x00000101/46265, Dst:0x00000101/340, Flags:None
+        p2 = re.compile(r'^\s*Src:[^\/]+\/(?P<src_sap>\d+), Dst:[^\/]+\/(?P<dst_sap>\d+), Flags:(?P<flags>.+)$')
+
+        #HA_SEQNO:0X00000000, RRtoken:0x0A8359C5, Sync:UNKNOWN, Payloadsize:308
+        p3 = re.compile(r'^\s*HA_SEQNO:(?P<ha_seqno>[^,]+), RRtoken:(?P<rr_token>[^,]+), Sync:(?P<sync>[^,]+), Payloadsize:(\d+)$')
+
+        for line in out.splitlines():
```
Comment (LukasMcClelland) [inline src/genie/libs/parser/nxos/show_hsrp_event_history.py:189]:
```suggestion
        for line in output.splitlines():
```

**Review by LukasMcClelland — CHANGES_REQUESTED** [review]
Thank you for opening up another PR with your contributions.

I caught an issue, but besides that it looks good

**Review by lsheikal — APPROVED** [review]

**Review by LukasMcClelland — APPROVED** [review]

**Pull request merged**